modules/core/src/lib/composite-layer.js

```diff
@@ -85,7 +86,7 @@ export default class CompositeLayer extends Layer {
     };
 
     if (sublayerProps) {
-      Object.assign(newProps, sublayerProps, {
+      Object.assign(newProps, sublayerProps, userProps && userProps[sublayerProps.id], {
```

Comment:
Good to add some comments here about what we are doing. I looked at this function the other day and it already took me a while to get oriented.